assignments/homework/chernyavskaya/hw5.py

```diff
@@ -0,0 +1,119 @@
+hdu_gaia = fits.open('astropy_notebooks/data/LMCDensFits1k.fits')[0]
+plt.imshow(hdu_gaia.data);
+plt.title("GAIA Data", fontsize=20)
+plt.show()
+p_init = models.Gaussian2D()
+y, x = np.mgrid[:750, :1000]
+p_fit = fitter(p_init, x, y, z=hdu_gaia.data)
+plt.imshow(p_fit(x, y));
+plt.title("2D Gaussian fit", fontsize=20)
+plt.show()
+plt.imshow(hdu_gaia.data - p_fit(x, y));
+plt.title("GAIA Data -  2D Gaussian fit", fontsize=20);
+plt.show()
+
+
+# Challenge
+# 1. Modify the plot of the combined fit to show the individual model components for the best-fit parameters.
+y_mod = y1 + 0.05 * x1**2 +0.2 * x1 + 2
+_ = plt.errorbar(x1, y_mod, yerr=yerr, fmt='.', c='blue')
+
+# fitting
+combined_init = models.Gaussian1D() + models.Polynomial1D(degree=2)
+combined_fit = fitter(combined_init, x1, y_mod, weights=1/yerr)
+
+plt.plot(xfine, combined_fit(xfine), lw=4, c='green', label='combined')
+plt.plot(xfine, combined_fit[0](xfine), lw=4, c='red', label='gaussian')
+plt.plot(xfine, combined_fit[1](xfine), lw=4, c='magenta', label='polynomial')
+plt.title("Combined fits separated", fontsize=20);
+plt.show()
+
+# 2. Continue the previous challenge to fit the LMC source density map by fitting two Gaussians to it. Be aware 
+# that especially with compound models, initial values matter! [advanced]
+print("I'm not sure how to set up the initial values, but the solution would be a combination of the two previous questions.")
+
+# Challenge
+# 1. Carry out aperture photometry using a radius of 5 and 10 pixels, and compare the results in a scatter plot
+star_image = load_star_image()
+sigma_clip = SigmaClip(sigma=3)
+bkg_estimator = MedianBackground()
+bkg = Background2D(star_image.data, (265, 265), sigma_clip=sigma_clip, bkg_estimator=bkg_estimator)
+star_image_nobkg = star_image.data - bkg.background
+mean, median, std = sigma_clipped_stats(star_image_nobkg, sigma=3)
+daofind = DAOStarFinder(fwhm=5, threshold=10*std)
+sources = daofind(star_image_nobkg)
+ap_5 = CircularAperture(list(zip(sources['xcentroid'], sources['ycentroid'])), r=5)
+ap_10 = CircularAperture(list(zip(sources['xcentroid'], sources['ycentroid'])), r=10)
+phot_5 = aperture_photometry(star_image_nobkg, ap_5)
+phot_10 = aperture_photometry(star_image_nobkg, ap_10)
+fig, ax = plt.subplots(2, figsize=(10, 15))
+ax[0].scatter(phot_5['xcenter'], phot_5['ycenter'], s=phot_5['aperture_sum']/10000, c='blue')
+ax[0].set_title("Photometry with aperture radius 5", fontsize=20);
+ax[1].scatter(phot_10['xcenter'], phot_10['ycenter'], s=phot_10['aperture_sum']/10000, c='red');
+ax[1].set_title("Photometry with aperture radius 10", fontsize=20)
+fig.show()
+
+# 2. Using this, construct a table containing just the sources where the 5 and 10 pixel fluxes agree to within a factor of 2
+flux_5 = phot_5['aperture_sum']
+flux_10 = phot_10['aperture_sum']
+tendiv5= flux_10/flux_5 < 2
+fivediv10 = flux_5/flux_10 < 2
+flux_factor2 = sources[tendiv5 & fivediv10]
```

Comment:
The aperture10 photometry is wrong is some cases (look at IDs{124, 160, 161, 185, 248, 251, 260, 1210, 1325, 1361, 1374}, the photometry is a negative number). So those should ideally not end up in the selection.
If you do the division only once, and check for both `<2` and `>0.5` then you would filter those out, too. And of course, you can then do the division only once, but that is very likely premature code optimization.